src/datta_lab_to_nwb/markowitz_gillis_nature_2023/optogeneticinterface.py

```diff
@@ -0,0 +1,104 @@
+"""Primary class for converting fiber photometry data (dLight fluorescence)."""
+# Standard Scientific Python
+import pandas as pd
+import numpy as np
+
+# NWB Ecosystem
+from pynwb.file import NWBFile
+from pynwb.core import DynamicTableRegion
+from pynwb.ophys import RoiResponseSeries
+from pynwb.ogen import OptogeneticSeries
+from neuroconv.basedatainterface import BaseDataInterface
+from neuroconv.utils import load_dict_from_file
+from neuroconv.tools import nwb_helpers
+from hdmf.backends.hdf5.h5_utils import H5DataIO
+
+# Local
+
+
+class OptogeneticInterface(BaseDataInterface):
+    """Optogenetic interface for markowitz_gillis_nature_2023 conversion"""
+
+    def __init__(self, file_path: str, session_uuid: str, metadata_path: str):
+        # This should load the data lazily and prepare variables you need
+        columns = (
+            "uuid",
+            "feedback_status",
+            "stim_duration",
+            "stim_frequency",
+            "power",
+            "area",
+            "timestamp",
+        )
+        super().__init__(
+            file_path=file_path,
+            session_uuid=session_uuid,
+            columns=columns,
+            metadata_path=metadata_path,
+        )
+
+    def get_metadata(self) -> dict:
+        metadata = super().get_metadata()
+        session_metadata = load_dict_from_file(self.source_data["metadata_path"])
+        session_metadata = session_metadata[self.source_data["session_uuid"]]
+        metadata["Subject"]["sex"] = session_metadata["sex"]
+        metadata["Optogenetics"]["area"] = session_metadata["optogenetic_area"]
+        metadata["Optogenetics"]["stim_frequency_Hz"] = session_metadata["stim_frequency_Hz"]
+        metadata["Optogenetics"]["pulse_width_s"] = session_metadata["pulse_width_s"]
+        metadata["Optogenetics"]["stim_duration_s"] = session_metadata["stim_duration_s"]
+        metadata["Optogenetics"]["power_watts"] = session_metadata["power_watts"]
+
+        return metadata
+
+    def get_metadata_schema(self) -> dict:
+        metadata_schema = super().get_metadata_schema()
+        metadata_schema["properties"]["Optogenetics"] = {
+            "type": "object",
+            "properties": {
+                "area": {"type": "string"},
+                "stim_frequency_Hz": {"type": "number"},
+                "stim_duration_s": {"type": "number"},
+                "power_watts": {"type": "number"},
+            },
+        }
+        return metadata_schema
+
+    def run_conversion(self, nwbfile: NWBFile, metadata: dict):
+        """Run conversion of data from the source file into the nwbfile."""
+        session_df = pd.read_parquet(
+            self.source_data["file_path"],
+            columns=self.source_data["columns"],
+            filters=[("uuid", "==", self.source_data["session_uuid"])],
+        )
+
+        device = nwbfile.create_device(
+            name="Opto Engine MRL-III-635",
```

Comment:
The very last correction to make in this PR - please remove spaces from object names (they become attributes or dict keys in the parent class objects so it just makes things weird)

```suggestion
            name="OptoEngineMRL",
```